assets/training/finetune_acft_hf_nlp/src/finetune/finetune.py

```diff
@@ -661,6 +662,32 @@ def resolve_deepspeed_config(args: Namespace) -> str:
     return default_deepspeed_config
 
 
+def update_deepspeed_config(ds_config_json: Dict[str, Any]) -> Optional[str]:
+    """Update the deepspeed config based on env flags."""
+    enable_profiler = os.environ.get("AZUREML_ENABLE_DEEPSPEED_PROFILER", False)
+    is_ds_config_updated = False
+    if enable_profiler and "flops_profiler" not in ds_config_json:
+        logger.info("Env variable is set to enable profiler.")
+        profiler_dict = dict(
+            flops_profiler=dict(
+                enabled=True,
+                profile_step=1,
+                module_depth=-1,
+                top_modules=1,
+                detailed=True,
+                output_file=None
+            )
+        )
+        ds_config_json.update(profiler_dict)
+        is_ds_config_updated = True
+
+    if is_ds_config_updated:
+        logger.info(f"Saving the updated deepspeed to new location: {DEFAULT_UPDATED_DEEPSPEED_PREFIX}")
+        with open(DEFAULT_UPDATED_DEEPSPEED_PREFIX, 'w', encoding="utf-8") as wptr:
+            json.dump(ds_config_json, wptr, indent=2)
+        return DEFAULT_UPDATED_DEEPSPEED_PREFIX
```

Comment:
can we use update_json_file_and_overwrite from azureml.acft.accelerator.code_utils?